### DISCUSSION TITLE
Conditional shell argument

### DISCUSSION THREAD

**Discussion by francium** [xonsh/xonsh#5015] [Q&A]
I've tried searching for "xonsh conditional argument" online, but I haven't found what I'm looking for.
If I've missed a similar discussion, please let me know.

My question can be summarized in this snippet,
```python
import sys

if len(sys.argv) == 2 and sys.argv[1] == '-p':
    git -p status
else:
    git status
``` 
I want to add flags to the shell command conditionally. Here I'm adding the `-p` flag conditionally.
This works, however I have to duplicate my shell command explicitly.

What I'd like to be able to do is,
```python
import sys

flag = ''
if len(sys.argv) == 2 and sys.argv[1] == '-p':
    flag = '-p'

git @(flag) status
```

However this does not work as you expect. Instead of omitting the argument if the `flag == ''`, a argument of `''` appears to be passed into the shell command, `['git', '', 'status']`.

I've also tried `flag = None`, but that does not work either, it passes `'None'` as a string to the shell command.

Is there a way of doing what I'm trying to do properly?

**Comment by anki-code** [comment]
You can use more universal way by using [`$args` in `ExecAlias`](https://github.com/anki-code/xonsh-cheatsheet#simple-aliases):

```python
aliases['gits'] = "git @($args) status"

$XONSH_TRACE_SUBPROC = True
gits -p
# TRACE SUBPROC: (['gits', '-p'],)
# TRACE SUBPROC: (['git', '-p', 'status'],)
```

More example in the [xonsh-cheatsheet](https://github.com/anki-code/xonsh-cheatsheet#simple-aliases).
